code/game/objects/items/devices/radio/encryptionkey.dm

```diff
@@ -22,6 +22,16 @@
 	channels = list("Syndicate" = 1)
 	origin_tech = "syndicate=1;engineering=3;bluespace=2"
 	syndie = 1 //Signifies that it de-crypts Syndicate transmissions
+	var/change_voice = 1
+	var/fake_name = "Agent ALERT_A_CODER"
+
+/obj/item/encryptionkey/syndicate/New()
```

Comment:
Also, rather than pick_n_take from the global list, just include this snippet
```
/obj/item/encryptionkey/syndicate
    var/static/list/fakename_list
/obj/item/encryptionkey/syndicate/Initialize()
    if(!LAZYLEN(fakename_list))
        fakename_list = GLOB.syndie_fakenames.Copy()
    . = ..()
    if(change_voice)
        fake_name = "Agent " + pick_n_take(fakename_list)
```

Reply:
Thanks for the tip with how to do it with a list copy.
However it does not work for me with Initialize(). I see the list created when viewing variables, but apparently the proc does not get called when I spawn in the encryption keys. New() works flawlessly though.

Reply:
Oh, it's fucking shit radio code. On top of Initialize, you either need to remove encryptionkey.dm#L16
Or add
```
/obj/item/encryptionkey/syndicate/New()
    . = ........()
```

Reply:
I got rid of encryptionkey.dm#L16 and it seems to work fine now.